### PR TITLE
Enables java tests to use the test helper for S3

### PR DESCRIPTION
**Pull request by TingDaoK**
*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/test/java/software/amazon/awssdk/crt/test/S3ClientTest.java [hunk 1]
```diff
@@ -1194,67 +1197,6 @@ public void onFinished(S3FinishedResponseContext context) {
         }
     }
 
-    @Test
-    public void testS3GetChecksums() {
```
Comment (TingDaoK) [inline src/test/java/software/amazon/awssdk/crt/test/S3ClientTest.java:1198]: it just repeats the same thing as the getting part of `testS3PutTrailerChecksums`, so, there is no reason to repeat it twice

File: README.md [hunk 2]
```diff
@@ -236,6 +238,22 @@ These can be set persistently via Maven settings (usually in `~/.m2/settings.xml
...
 
+### S3 Tests
+
+To run `S3ClientTest` successfully, aws credentials and a set of pre-configured S3 buckets will be needed. There is a helper script from aws-c-s3 that can be used to set up the test environment, [here](https://github.com/awslabs/aws-c-s3/tree/main/tests/test_helper).
+
+To run the tests properly:
+
+``` sh
+cd aws-crt-java
+pip3 install boto3
```
Comment (graebm) [inline README.md:249]: trivial: it's not installed as `pip3` on every platform. The most consistent way is:
```suggestion
python3 -m pip install boto3
```

File: README.md [hunk 3]
```diff
@@ -196,6 +196,8 @@ static: certificate has an existing certificate-key pair that was previously imp
 ## Testing
 Many tests require environment variables to be set. These environment variables are translated at runtime to system properties for use by the tests. These tests will be quietly skipped if the properties they require are not set.
 
+### MQTT Tests
+
```
Comment (graebm) [inline README.md:200]: The env vars listed below aren't ONLY about MQTT
```suggestion
```

You could break up the list below by topic, like it is in [aws-crt-python's README](https://github.com/awslabs/aws-crt-python/blob/main/guides/dev/README.md), and include the S3 test_helper.py directions as a sub-bullet-point of the S3 topic

File: src/test/java/software/amazon/awssdk/crt/test/S3ClientTest.java [hunk 4]
```diff
@@ -50,21 +50,20 @@
...
-            ? "crts-east1--use1-az4--x-s3.s3express-use1-az4.us-east-1.amazonaws.com"
-            : System.getenv("S3EXPRESS_ENDPOINT_EAST1");
-    static final String REGION = System.getenv("REGION") == null ? "us-west-2" : System.getenv("REGION");
-
-
-    static final String COPY_SOURCE_BUCKET = "aws-crt-test-stuff-us-west-2";
-    static final String COPY_SOURCE_KEY = "crt-canary-obj.txt";
-    static final String X_AMZ_COPY_SOURCE_HEADER = "x-amz-copy-source";
+    static final String BUCKET_NAME = System.getenv("CRT_S3_TEST_BUCKET_NAME") == null
+            ? "aws-c-s3-test-bucket"
```
Comment (graebm) [inline src/test/java/software/amazon/awssdk/crt/test/S3ClientTest.java:54]: We should probably just skip the S3 tests if this isn't set. That's how it is with all our other testing environment variables. It lets 3rdparty people run basic tests without creating an AWS account and putting stuff in it

File: src/test/java/software/amazon/awssdk/crt/test/S3ClientTest.java [hunk 5]
```diff
@@ -1453,6 +1395,11 @@ public void testS3Copy() {
         skipIfNetworkUnavailable();
         Assume.assumeTrue(hasAwsCredentials());
 
+        String COPY_SOURCE_BUCKET = "aws-crt-test-stuff-us-west-2";
```
Comment (graebm) [inline src/test/java/software/amazon/awssdk/crt/test/S3ClientTest.java:1398]: this is still hard-coded though?

Comment (TingDaoK) [reply src/test/java/software/amazon/awssdk/crt/test/S3ClientTest.java:1398]: https://github.com/awslabs/aws-crt-java/pull/788/files#diff-61fd3ff6b0c1bb48020cafcdd8ae2a00f7ba2e2da6aff15d125868eda22bdc96R1390-R1393

The copy tests are disabled

File: src/test/java/software/amazon/awssdk/crt/test/S3ClientTest.java (hunk 4 above)
Comment (TingDaoK) [reply src/test/java/software/amazon/awssdk/crt/test/S3ClientTest.java:54]: we already skip those s3 tests if there is no AWS credentials set in the environment, which allows 3rdparty people to run the other tests properly.  

It seems an annoy extra step to setup an environment variable to run the tests to me.

File: README.md [hunk 6]
```diff
@@ -193,25 +193,22 @@ Please note that on Mac, once a private key is used with a certificate, that cer
 static: certificate has an existing certificate-key pair that was previously imported into the Keychain.  Using key from Keychain instead of the one provided.
 ```
 
-## Testing
-Many tests require environment variables to be set. These environment variables are translated at runtime to system properties for use by the tests. These tests will be quietly skipped if the properties they require are not set.
+## Tests with resource required
```
Comment (graebm) [inline README.md:196]: trivial: this sounds weird, the old simple way was fine
```suggestion
## Testing
```

File: README.md [hunk 7]
```diff
@@ -18,7 +18,7 @@ This library is licensed under the Apache 2.0 License.
 * [Platform-Specific JARs](#platform-specific-jars)
 * [System Properties](#system-properties)
 * [TLS Behavior](#tls-behavior)
-* [Testing](#testing)
+* [Tests with resource required](#tests-with-resource-required)
```
Comment (graebm) [inline README.md:21]: trivial: this sounds weird, the old simple way was fine
```suggestion
* [Testing](#testing)
```

File: README.md [hunk 8]
```diff
@@ -236,6 +233,34 @@ These can be set persistently via Maven settings (usually in `~/.m2/settings.xml
...
+Most of proxy related tests need pre-configured Proxy host to run the tests properly.
+
+* Required environment variables:
+  * `AWS_TEST_HTTP_PROXY_HOST`: Hostname of proxy
+  * `AWS_TEST_HTTP_PROXY_PORT`: Port of proxy
+  * `NETWORK_TESTS_DISABLED`: Set this if tests are running in a constrained environment where network access is not guaranteed/allowed.
+
+### S3 Tests
+
+Most of S3 related tests requires aws credentials and a set of pre-configured S3 buckets. There is a helper script from aws-c-s3 that can be used to set up the test environment, [here](https://github.com/awslabs/aws-c-s3/tree/main/tests/test_helper).
```
Comment (graebm) [inline README.md:247]:
```suggestion
Most of S3 related tests require AWS credentials and a set of pre-configured S3 buckets. There is a helper script from aws-c-s3 that can be used to set up the test environment, [here](https://github.com/awslabs/aws-c-s3/tree/main/tests/test_helper).
```

**Review by graebm — APPROVED** [review]

**Pull request merged**